src/com/maddyhome/idea/vim/command/CommandState.kt

```diff
@@ -319,10 +339,10 @@ class CommandState private constructor() {
   }
 
   enum class SubMode {
-    NONE, SINGLE_COMMAND, REGISTER_PENDING, VISUAL_CHARACTER, VISUAL_LINE, VISUAL_BLOCK
+    NONE, SINGLE_COMMAND, REGISTER_PENDING, REPLACE_CHARACTER, VISUAL_CHARACTER, VISUAL_LINE, VISUAL_BLOCK
```

Comment:
Yes, that's it. 

I introduced `REPLACE_CHARACTER` because I don't think this is a good fit for the existing `REPLACE` mode. It's tracking the state of an incomplete command, rather than actually switching into `REPLACE` mode. It doesn't feel right to do a full mode switch in the middle of command entry - we're not actually in replace mode, we're waiting for an argument to a command that will do a replace. We want to look like we're in replace mode, but not actually be in it.

And pragmatically, if we used `REPLACE`, we'd still need another piece of state to know when to pop the `REPLACE` mode if hitting `<Esc>` or entering an incorrect digraph. E.g. `handleDigraph` doesn't currently pop modes, so if we're entering a digraph, we don't know if we're entering it in actual replace mode, or as the argument for an incomplete action. The sub mode gives us everything we need, and tells us we're still in `COMMAND` or `VISUAL` mode (which we are, waiting for a command to complete), and gives us something we can update the caret with.

Reply:
Okay, thank you for explanation. I have some thoughts about removing `subMode` at all because it doesn't exist in vim, but it might be a future refactoring. Let's use our current structure.